recipes/qt/6.x.x/conanfile.py

```diff
@@ -0,0 +1,541 @@
+            self.requires("double-conversion/3.1.5")
+        if self.options.get_safe("with_freetype", False) and not self.options.multiconfiguration:
+            self.requires("freetype/2.10.4")
+        if self.options.get_safe("with_fontconfig", False):
+            self.requires("fontconfig/2.13.92")
+        if self.options.get_safe("with_icu", False):
+            self.requires("icu/68.2")
+        if self.options.get_safe("with_harfbuzz", False) and not self.options.multiconfiguration:
+            self.requires("harfbuzz/2.7.4")
+        if self.options.get_safe("with_libjpeg", False) and not self.options.multiconfiguration:
+            if self.options.with_libjpeg == "libjpeg-turbo":
+                self.requires("libjpeg-turbo/2.0.6")
+            else:
+                self.requires("libjpeg/9d")
+        if self.options.get_safe("with_libpng", False) and not self.options.multiconfiguration:
+            self.requires("libpng/1.6.37")
+        if self.options.with_sqlite3 and not self.options.multiconfiguration:
+            self.requires("sqlite3/3.33.0")
+            self.options["sqlite3"].enable_column_metadata = True
+        if self.options.get_safe("with_mysql", False):
+            self.requires("libmysqlclient/8.0.17")
+        if self.options.with_pq:
+            self.requires("libpq/12.2")
+        if self.options.with_odbc:
+            if self.settings.os != "Windows":
+                self.requires("odbc/2.3.7")
+        if self.options.gui and self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("xorg/system")
+            if not tools.cross_building(self, skip_x64_x86=True):
+                self.requires("xkbcommon/1.0.3")
+        if self.options.get_safe("opengl", "no") != "no":
+            self.requires("opengl/system")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.8")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        shutil.move("qt-everywhere-src-%s" % self.version, "qt6")
+
+        tools.replace_in_file(os.path.join("qt6", "CMakeLists.txt"),
+                        "enable_testing()",
+                        "include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)\nconan_basic_setup(KEEP_RPATHS)\n"
+                               "set(QT_EXTRA_INCLUDEPATHS ${CONAN_INCLUDE_DIRS})\n"
+                               "set(QT_EXTRA_DEFINES ${CONAN_DEFINES})\n"
+                               "set(QT_EXTRA_LIBDIRS ${CONAN_LIB_DIRS})\n"
+                               "enable_testing()")
+
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        tools.replace_in_file(os.path.join("qt6", "qtbase", "cmake", "QtInternalTargets.cmake"),
+                              "target_compile_options(PlatformCommonInternal INTERFACE -Zc:wchar_t)",
+                              "target_compile_options(PlatformCommonInternal INTERFACE -Zc:wchar_t -Zc:twoPhase-)")
```

Comment:
patching in source method 🙊

Reply:
this is mandatory because of [`no_copy_source`](https://github.com/conan-io/conan-center-index/pull/4412/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R68). I'm not sure if we still need this attribute though. I'll check it.